src/promptflow-devkit/promptflow/_cli/_pf/_service.py

```diff
@@ -202,6 +202,13 @@ def validate_port(port, force_start):
             sys.stdout = old_stdout
             sys.stderr = old_stderr
     else:
+        # Add executable script dir to PATH to make sure the subprocess can find the executable, especially in notebook
+        # environment which won't add it to system path automatically.
+        python_dir = os.path.dirname(sys.executable)
+        executable_dir = os.path.join(python_dir, "Scripts") if platform.system() == "Windows" else python_dir
+        if executable_dir not in os.environ["PATH"].split(os.pathsep):
+            os.environ["PATH"] = executable_dir + os.pathsep + os.environ["PATH"]
```

Comment:
shall this be added to head or end of PATH list?

Reply:
head